### PR TITLE
7903693 Add contribution guidelines section

### PR DESCRIPTION
**Pull request by LudwikJaniuk**
Adding in `building.md` as it seems the place one would read if making changes.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [CODETOOLS-7903693](https://bugs.openjdk.org/browse/CODETOOLS-7903693): Add contribution guidelines section (**Enhancement** - P4)

### Reviewers
 * [Jonathan Gibbons](https://openjdk.org/census#jjg) (@jonathan-gibbons - **Reviewer**) ⚠️ Review applies to [4233dae7](https://git.openjdk.org/jtreg/pull/192/files/4233dae70835ef02ebd82ec1a88f6e1b9d7b9695)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jtreg.git pull/192/head:pull/192` \
`$ git checkout pull/192`

Update a local copy of the PR: \
`$ git checkout pull/192` \
`$ git pull https://git.openjdk.org/jtreg.git pull/192/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 192`

View PR using the GUI difftool: \
`$ git pr show -t 192`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jtreg/pull/192.diff">https://git.openjdk.org/jtreg/pull/192.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jtreg/pull/192#issuecomment-1988366694)

### CODE REVIEW

File: doc/building.md [hunk 1]
```diff
@@ -139,3 +139,16 @@ to create a dummy X server, the server will be terminated when the test is
...
+
+Contributors are encouraged to follow code style conventions in [Java Style Guidelines](https://cr.openjdk.org/~alundblad/styleguide/index-v6.html) 
+where reasonable. jtreg's command-line flags have a certain style due to their
+age, but new flags should strive to follow [JEP 293: Guidelines for JDK Command-Line Tool Options](https://openjdk.org/jeps/293). 
+One notable exception is the rule about case-sensitivity - jtreg's command line
+flags are case insensitive and due to backwards compatibility it seems 
+reasonable to keep this going forward.
+
+The jtreg codebase is very dependend on (jtharness)[https://github.com/openjdk/jtharness]. 
+The two respos should most often be viewed together. This also places constraints
```
Comment (jonathan-gibbons) [inline doc/building.md:153]: typo: `respos`

File: doc/building.md [hunk 2]
```diff
@@ -139,3 +139,16 @@ to create a dummy X server, the server will be terminated when the test is
...
 The logic for using VNC is encapsulated within the script _make/display.sh_.
+
+# Contribution guidelines
+
+Contributors are encouraged to follow code style conventions in [Java Style Guidelines](https://cr.openjdk.org/~alundblad/styleguide/index-v6.html) 
+where reasonable. jtreg's command-line flags have a certain style due to their
+age, but new flags should strive to follow [JEP 293: Guidelines for JDK Command-Line Tool Options](https://openjdk.org/jeps/293). 
+One notable exception is the rule about case-sensitivity - jtreg's command line
+flags are case insensitive and due to backwards compatibility it seems 
+reasonable to keep this going forward.
```
Comment (jonathan-gibbons) [inline doc/building.md:150]: Yes, we should specify the case we encourage folk to use, but permit case-insensitive matching.

Comment (jonathan-gibbons) [inline doc/building.md:150]: Generally, it is good practice to not use time-dependent text, like "going forward", in any spec. 

I would replace the entire 148-150 lines with,
```
For backwards compatibility, `jtreg` option names are case-insensitive.
```

**Review by jonathan-gibbons — APPROVED** [review]
Thanks for the contribution.

One suggestion and one typo. Please fix before integrating.

File: doc/building.md (hunk 2 above)
Comment (LudwikJaniuk) [reply doc/building.md]: Good advice, I've adopted this language.

**Pull request closed without merging**